tensordict/tensordict.py

```diff
@@ -2149,7 +2158,9 @@ def unlock(self):
         self._is_locked = False
         self._is_shared = False
         self._is_memmap = False
-        for key in self.keys():
+        keys_view = _TensorDictKeysView(tensordict=self, include_nested=True, leaves_only=False,
+                                        error_on_loop=False)
+        for key in keys_view:
             if is_tensordict(self.entry_class(key)):
                 self.get(key).unlock()
```

Comment:
Ditto here, we're doing extra work by recursing over values with both the keys view and with recursive calls to `unlock`. I think we should change

```diff
- self.get(key).unlock()
+ value = self.get(key)
+ value._is_locked = False
+ value._is_shared = False
+ value._is_memmap = False
```